R/zchunk_L203.demand_input.R

```diff
@@ -88,288 +93,572 @@ module_aglu_L203.demand_input_DISABLED <- function(command, ...) {
+      # Combine crop and meat food demand in Pcal
+      bind_rows(L105.an_Food_Pcal_R_C_Y) %>%
+      filter(year %in% aglu_demand_calyears) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      L203.ag_an_Food_Pcal_R_C_Y
+
+    A_demand_technology_R_Yh %>%
+      # Select food demand
+      filter(supplysector %in% c("FoodDemand_Crops", "FoodDemand_Meat")) %>%
+      # Map in food demand by region / commodity / year
+      left_join_error_no_match(L203.ag_an_Food_Pcal_R_C_Y, by = c("region", "technology" = "GCAM_commodity", "year")) %>%
+      mutate(calOutputValue = round(value, aglu.DIGITS_CALOUTPUT),
+             share.weight.year = year,
+             # Subsector and technology shareweights (subsector requires the year as well)
+             subs.share.weight = if_else(calOutputValue > 0, 1, 0),
+             tech.share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["StubTechProd"]])) %>%
+      filter(!region %in% aglu.NO_AGLU_REGIONS) %>%           # Remove any regions for which agriculture and land use are not modeled
+      filter(year %in% BASE_YEARS) ->                         # ALSO SUBSET THE CALIBRATION TABLES TO ONLY THE MODEL BASE YEARS
+      L203.StubTechProd_food
+
+    # Build L203.StubTechProd_nonfood_crop and L203.StubTechProd_nonfood_meat: crop and meat nonfood demand by technology and region
+    L109.ag_ALL_Mt_R_C_Y %>%
+      # Combine the balance tables of crop and meat in Mt
+      bind_rows(L109.an_ALL_Mt_R_C_Y) %>%
+      filter(year %in% aglu_demand_calyears) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      select(GCAM_region_ID, region, GCAM_commodity, year, OtherUses_Mt, NetExp_Mt) -> # only nonfood and net exports data will be used
+      L203.ag_an_ALL_Mt_R_C_Y
+
+    A_demand_technology_R_Yh %>%
+      # Select nonfood demand
+      filter(supplysector %in% c("NonFoodDemand_Crops", "NonFoodDemand_Meat")) %>%
+      # Map in nonfood demand by region / commodity / year
+      left_join_error_no_match(L203.ag_an_ALL_Mt_R_C_Y, by = c("region", "technology" = "GCAM_commodity", "year")) %>%
+      mutate(calOutputValue = round(OtherUses_Mt, aglu.DIGITS_CALOUTPUT),
+             share.weight.year = year,
+             # Subsector and technology shareweights (subsector requires the year as well)
+             subs.share.weight = if_else(calOutputValue > 0, 1, 0),
+             tech.share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["StubTechProd"]])) %>%
+      filter(!region %in% aglu.NO_AGLU_REGIONS) %>%           # Remove any regions for which agriculture and land use are not modeled
+      filter(year %in% BASE_YEARS) ->                         # ALSO SUBSET THE CALIBRATION TABLES TO ONLY THE MODEL BASE YEARS
+      L203.StubTechProd_nonfood
+
+    # Build L203.StubTechFixOut_exp: animal exports for net exporting regions in all periods
```

Comment:
Maybe note that we hold these constant in the future and that is why you are copying the final base year to all future years.